src/Bridge/Doctrine/DBAL/Types/AbstractEnumType.php

```diff
@@ -14,12 +14,17 @@
 use Doctrine\DBAL\Types\Type;
 use Elao\Enum\EnumInterface;
 
+/**
+ * @template T
+ */
 abstract class AbstractEnumType extends Type
 {
     /**
      * The enum FQCN for which we should make the DBAL conversion.
      *
-     * @return class-string<EnumInterface>
+     * @return string&EnumInterface
+     * @phpstan-return class-string<EnumInterface<T>>
+     * @psalm-return class-string<EnumInterface<T>>
```

Comment:
I don't think we should duplicate these annotations here. AFAIK PHPStan parses `@psalm-*` annotations too so we can just stick to psalm ones.

Reply:
I did not know :+1: Thanks I will try.

Reply:
Do you have any reference ? It looks to work indeed :+1: